### PR TITLE
TOURCMS-10026 modify login so that we use apikey from script to creat…

### PR DESCRIPTION
**Pull request by palisis-juancarlos**
…e tourcms agent

### CODE REVIEW

File: config.php [hunk 1]
```diff
@@ -16,3 +16,4 @@
 $BASE_URL = 'http://api.tourcms.local';
 $MARKETPLACE_ID_FOR_OPERATOR = 0;
 $SCRIPT_LOGIN = 'http://localhost/admin.tourcms.com/partner/api_login_new.php';
+$SCRIPT_LOGIN_AGENT = 'http://localhost/admin.tourcms.com/partner/app_api_login_new.php';
```
Comment (palisis-juanra) [inline config.php]: pleae,if first one not used. Remove it

File: config.php-default [hunk 2]
```diff
@@ -16,3 +16,4 @@ $TIMEOUT = ;
 $BASE_URL = '';
 $MARKETPLACE_ID_FOR_OPERATOR = '' ;
 $SCRIPT_LOGIN = '';
+$SCRIPT_LOGIN_AGENT = '';
```
Comment (palisis-juanra) [inline config.php-default]: Please, if first one not used. Remove it

File: controllers/GeneralService.php [hunk 3]
```diff
@@ -1,8 +1,22 @@
 <?php
+use onboarding\services\RedisService;
 
 class GeneralService
 {
-    function getXMLFromValidation($url, $paramsFromGet)
+    static $cacheName;
```
Comment (palisis-juanra) [inline controllers/GeneralService.php]: This is more a const than a static field.
In addition, try to think a better name ... what do you think about agentApiKeyRedisKey?

File: controllers/GeneralService.php [hunk 4]
```diff
@@ -29,4 +43,57 @@ function getXMLFromValidation($url, $paramsFromGet)
 
         return $xml;
     }
+
+    public function getXMLForAgent($url, $paramsForPost)
```
Comment (palisis-juanra) [inline controllers/GeneralService.php]: This function is not about getting a XML for agent, its more a Curl post request ...

**Review by palisis-juanra — CHANGES_REQUESTED** [review]

File: config.php (hunk 1 above)
Comment (palisis-juancarlos) [reply config.php]: fixed

File: config.php-default (hunk 2 above)
Comment (palisis-juancarlos) [reply config.php-default]: fixed

File: controllers/GeneralService.php (hunk 4 above)
Comment (palisis-juancarlos) [reply controllers/GeneralService.php]: fixed

**Review by palisis-juanra — APPROVED** [review]

**Pull request merged**